js/interface.js

```diff
@@ -261,35 +280,180 @@ function getFolderContentsById(id, type, isSearchNav) {
   });
 }
 
+function restoreTrashItems(items) {
+  completedItems = 0;
+
+  $(items).each(function() {
+    var $element = $(this);
+    var itemID = $element.attr('data-id');
+    var itemName = $element.attr('data-name');
+    var restorePromise;
+
+    showSpinner(true);
+
+    $element.addClass('restore-fade');
+
+    if ($element.attr('data-file-type') === 'folder') {
+      restorePromise = Fliplet.API.request({
+        url: 'v1/media/folders/' + itemID + '/restore',
+        method: 'POST',
+      }).then(function() {
+        $element.remove();
+        checkboxStatus();
+        completedItems++;
+
+        currentFolders = currentFolders.filter(function(folder){
+          return folder.id != itemID;
+        });
+        $('.file-table-header input[type="checkbox"]').prop('checked', false);
+      })
+    } else {
+      restorePromise = Fliplet.API.request({
+        url: 'v1/media/files/' + itemID + '/restore',
+        method: 'POST',
+      }).then(function() {
+        $element.remove();
+        checkboxStatus();
+        completedItems++;
+
+        currentFiles = currentFiles.filter(function(file){
+          return file.id != itemID;
+        });
+
+        $('.file-table-header input[type="checkbox"]').prop('checked', false);
+      })
+    }
+    restorePromise.then(function(result) {
```

Comment:
This PR has a lot of the same problems mentioned in ttps://github.com/Fliplet/fliplet-widget-data-source/pull/101#discussion_r494517387 @squallstar Can you speak with @sofiiakvasnevska to verify induction, linting tools and internal code review are carried out correctly for new developers?

```suggestion

    restorePromise.then(function(result) {
```